iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -260,7 +261,7 @@ def set_node_index(
         c.dim: c for c in constraints if isinstance(c, WorkgroupConstraint)
     }
     other_constraints = [
-        c for c in constraints if not isinstance(c, HardwareConstraint)
+        c for c in constraints if not isinstance(c, (HardwareConstraint, Assumption))
```

Comment:
BTW, how is the assumption being checked/asserted? .apply?

Reply:
Right now, the assumption can be any sympy expression because it is only used to answer queries. So if there are errors in the assumption, they would come from sympy's error checking. But as long as it is a valid sympy expression it should be a valid assumption.